sdk/core/azure-core/src/http/http.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/http/http.hpp"
+
+bool Azure::Core::Http::Details::InsertHeaderWithValidation(
+    std::map<std::string, std::string>& headers,
+    std::string const& headerName,
+    std::string const& headerValue,
+    size_t* const invalidHeaderIndex)
+{
+  // Static table for validating header names. It is created just once for the program and reused
+  // each time AddHeader is called
+  static const uint8_t validChars[256] = {
```

Comment:
This table is not very readable. I would suggest to write it as switch/case, or some other more compact way. Maybe a combination of `if-else` and `switch-case`. Or, some table that can fit on one screen, and is readable.

Reply:
Although I understand, it works. But I think, it fyou have a bug here, it will be very difficult to spot. Even to figure out when reading, what is good and what is not, is a problem.

Reply:
IMHO, this is more readable than this https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/src/strings.cpp#L50, for instance. Even it can be seen in a one single screen, there's no quick understanding on what char is each position.
I prefer a longer list where can see human information about if